src/Compilers/Core/Portable/SourceFileResolver.cs

```diff
@@ -87,14 +146,17 @@ public override bool Equals(object obj)
             }
 
             var other = (SourceFileResolver)obj;
-            return string.Equals(_baseDirectory, other._baseDirectory, StringComparison.Ordinal) &&
-                _searchPaths.SequenceEqual(other._searchPaths, StringComparer.Ordinal);
+            return
+                string.Equals(_baseDirectory, other._baseDirectory, StringComparison.Ordinal) &&
+                _searchPaths.SequenceEqual(other._searchPaths, StringComparer.Ordinal) &&
+                (_pathMap.IsDefaultOrEmpty ? other._pathMap.IsDefaultOrEmpty : _pathMap.SequenceEqual(other._pathMap));
         }
 
         public override int GetHashCode()
         {
             return Hash.Combine(_baseDirectory != null ? StringComparer.Ordinal.GetHashCode(_baseDirectory) : 0,
-                   Hash.CombineValues(_searchPaths, StringComparer.Ordinal));
+                   Hash.Combine(Hash.CombineValues(_searchPaths, StringComparer.Ordinal),
+                   Hash.CombineValues(_pathMap)));
```

Comment:
Historically, not providing efficient implementation of Equals/GetHashCode for structures was a common source of perf issues in our components. Equals, and especially GetHashCode, should be as efficient as possible. I think that, since you've chosen to use an inefficient implementation, the burden of proof is on you, or you could simply provide/use an efficient implementation. If this code is executed only in specific, very constraint scenarios, doesn't affect behavior of any public API, and we, as a team, are fine to sacrifice performance due to some good reason, it is probably fine to leave it as is. However, you should provide the analysis and the rational behind this choice.